pkg/pillar/cmd/zedkube/aitoapiserver.go

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) 2024 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//go:build kubevirt
+
+package zedkube
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/lf-edge/eve/pkg/pillar/base"
+	"github.com/lf-edge/eve/pkg/pillar/kubeapi"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	"github.com/sirupsen/logrus"
+	"github.com/vishvananda/netlink"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// checkIoAdapterEthernet - check and create NAD for direct-attached ethernet
+func checkIoAdapterEthernet(ctx *zedkubeContext, aiConfig *types.AppInstanceConfig) error {
+
+	if aiConfig.FixedResources.VirtualizationMode != types.NOHYPER {
+		return nil
+	}
+	ioAdapter := aiConfig.IoAdapterList
+	for _, io := range ioAdapter {
+		if io.Type == types.IoNetEth {
+			nadname := "host-" + io.Name
+			_, ok := ctx.networkInstanceStatusMap.Load(nadname)
+			if !ok {
+				bringupInterface(io.Name)
```

Comment:
If we fail to prepare interface for assignment, shouldn't we propagate the error to AppInstanceStatus (and potentially stop the application from being deployed)? Seems we only log errors here.

Reply:
+1

Reply:
The problem is how to bubble up this error into AppInstanceStatus, when we launch the pod in domainmgr, get will get an error there, since the net-attach-def of this does not exist. We'll get the error to the user this way.

Reply:
Yes, but the error message will be something else and not useful for troubleshooting.

Reply:
Ok, I can let zedkube in the error condition to publish the IOAdapter status error. and zedmanager listening on this, and it can include this into the AppInatnceStatus error portion. Does this sound reasonable?

Reply:
actually I don't need to do that. I can do this in the domainmgr, during the pod configure setup stage, query for the net-attach-def for this ether port, if kubernetes side does not have it, it can generate this specific error condition for the UI.

Reply:
Since the domainmgr side the of kubevirt patch has not been made into the mainline, we'll change to add that error condition for checking the io-ethernet from k3s and express error over there.